core/retries/src/test/java/software/amazon/awssdk/retries/internal/RetryStrategyCircuitBreakerRemembersStateTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.retries.internal;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import software.amazon.awssdk.retries.api.BackoffStrategy;
+import software.amazon.awssdk.retries.api.RetryStrategy;
+import software.amazon.awssdk.retries.internal.RetryStrategyCommonTest.TestCase;
+
+/**
+ * Tests that the configured circuit breaker for each of the strategies remembers
+ * state across requests.
+ */
+public class RetryStrategyCircuitBreakerRemembersStateTest {
```

Comment:
This test was also added to replace [StandardRetryStrategyMiscTest](https://github.com/aws/aws-sdk-java-v2/blob/feature/master/sra-retries/core/retries/src/test/java/software/amazon/awssdk/retries/internal/StandardRetryStrategyMiscTest.java) as this behavior is expected for all the strategies.